src/hnsw/build.c

```diff
@@ -161,8 +250,11 @@ static void InitBuildState(HnswBuildState *buildstate, Relation heap, Relation i
     buildstate->columnType = GetIndexColumnType(index);
     buildstate->dimensions = GetHnswIndexDimensions(index);
 
+    // If a dimension wasn't specified try to infer it
+    if(buildstate->columnType == REAL_ARRAY || buildstate->columnType == INT_ARRAY)
+        if(buildstate->dimensions < 1) buildstate->dimensions = InferDimension(heap, indexInfo);
```

Comment:
We should always use braces `{,}` with `if`s and other basic blocs. 
It is particularly important for cases like these where you have multiple `if`s
 
I think the code above can be clarified as well. 
What happens when `buildstate->columnType` is not `REAL_ARRAY` or `INT_ARRAY`?

Reply:
Ah you're right, the operators aren't defined, good catch